mods/farming/api.lua

```diff
@@ -47,6 +47,44 @@ function farming.dirt_after_dig(pos, node, oldmetadata, digger)
 	minetest.swap_node(pos, {name = regN[node.name].soil.dry})
 end
 
+function farming.scythe_on_use(itemstack, user, pointed_thing)
+	if user == nil or not user:is_player() or pointed_thing.type ~= "node" then
+		return 
+	end
+	local pos = pointed_thing.under
+	local radius = itemstack:get_definition().radius
+	local efficacy = itemstack:get_definition().efficacy
+
+	-- works in a 2*radius square
+	for i=-radius,radius do
+		for j=-radius,radius do
+			local p = {x = pos.x + i, y = pos.y, z = pos.z + j}
+			local n = minetest.get_node(p)
+
+			-- dig crops if a grass species
+			if n ~= nil and minetest.get_item_group(n.name, "poaceae") > 0 and math.random() < efficacy then
+				if not minetest.dig_node(p, n, user) then -- could not dig
+					goto continue
```

Comment:
The general consensus was that we shouldn't use this idiom, since normal Lua 5.1 won't be able to run it. (blame me for using the idiom in the first place 🤣)